ignite/contrib/metrics/fid.py

```diff
@@ -0,0 +1,95 @@
+import os
+from typing import Callable, Sequence, Union
+
+import torch
+from torchvision import transforms
+
+from ignite.exceptions import NotComputableError
+from ignite.metrics.metric import Metric, reinit__is_reduced, sync_all_reduce
+
+try:
+    from scipy import linalg
+except ImportError:
+    raise RuntimeError("This contrib module requires scipy to be installed.")
+
+
+__all__ = ["FID"]
+
+
+class FID(Metric):
+    """Calculates FID metric
+
+    """
+
+    def __init__(
+        self, output_transform: Callable = lambda x: x, fid_model=None,
+    ):
+        if fid_model is None:
+            try:
+                from torchvision import models
+
+                fid_model = models.inception_v3(pretrained=True)
+            except ImportError:
+                raise ValueError("Argument fid_model should be set")
+
+        self._device = torch.device("cuda" if torch.cuda.is_available() else "cpu")
+        self._fid_model = fid_model.eval().to(self._devide)
+
+    @reinit__is_reduced
+    def reset(self) -> None:
+        self._num_of_data = 0
+
+        self._features_sum_real = None
```

Comment:
Here, IMO, we can set all those things as `torch.tensor(0, device=self.device, dtype=torch.float32)` and we can simply use `+= batch_features_real` etc ...